Libraries/MLXVLM/Models/Idefics3.swift

```diff
@@ -871,3 +886,332 @@ public class Idefics3Processor: UserInputProcessor {
         }
     }
 }
+
+// MARK: - SmolVLMProcessor and configuration
+
+public struct SmolVLMProcessorConfiguration: Codable, Sendable {
+    public struct Size: Codable, Sendable {
+        public let longestEdge: Int
+        enum CodingKeys: String, CodingKey {
+            case longestEdge = "longest_edge"
+        }
+    }
+
+    public struct VideoSampling: Codable, Sendable {
+        public let fps: Int
+        public let maxFrames: Int
+        // Intentionally ignoring videoSize because I believe it's still wrong in the config files
+        //        public let videoSize: Size
+
+        enum CodingKeys: String, CodingKey {
+            case fps
+            case maxFrames = "max_frames"
+        }
+    }
+
+    public let imageMean: [CGFloat]
+    public let imageStd: [CGFloat]
+    public let size: Size
+    public let maxImageSize: Size
+    public let videoSampling: VideoSampling
+    private let _imageSequenceLength: Int?
+    // TODO: this does not come in preprocessor_config.json, verify where transformers gets it from
+    public var imageSequenceLength: Int { _imageSequenceLength ?? 64 }
+
+    init(
+        imageMean: [CGFloat], imageStd: [CGFloat], size: Size, maxImageSize: Size,
+        videoSampling: VideoSampling, imageSequenceLength: Int?
+    ) {
+        self.imageMean = imageMean
+        self.imageStd = imageStd
+        self.size = size
+        self.maxImageSize = maxImageSize
+        self.videoSampling = videoSampling
+        self._imageSequenceLength = imageSequenceLength
+    }
+
+    public var imageMeanTuple: (CGFloat, CGFloat, CGFloat) {
+        (imageMean[0], imageMean[1], imageMean[2])
+    }
+    public var imageStdTuple: (CGFloat, CGFloat, CGFloat) {
+        (imageStd[0], imageStd[1], imageStd[2])
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case imageMean = "image_mean"
+        case imageStd = "image_std"
+        case size
+        case maxImageSize = "max_image_size"
+        case videoSampling = "video_sampling"
+        case _imageSequenceLength = "image_seq_len"
+    }
+}
+
+public class SmolVLMProcessor: UserInputProcessor {
+    private let config: SmolVLMProcessorConfiguration
+    private let tokenizer: any Tokenizer
+
+    // FIXME: hardcoded values for now
+
+    // Hardcode this since we can't pass it in or rely on it from the preprocessor config.
+    let imageTokenId = 49190
+    let imageToken = "<image>"
+    let fakeImageToken = "<fake_token_around_image>"
+    let globalImageToken = "<global-img>"
+
+    var maxProcessingImageSize: CGFloat { CGFloat(config.size.longestEdge) }  // 2048
+    var fixedImageSize: CGFloat { CGFloat(config.maxImageSize.longestEdge) }  // 384 for big models, 512 for small models (200-500M)
+    var imageSequenceLength: Int { config.imageSequenceLength }
+    var maxVideoFrames: Int { 20 /*config.videoSampling.maxFrames*/ }
```

Comment:
```suggestion
    var maxVideoFrames: Int { 20 /*config.videoSampling.maxFrames*/ }    // Limited to reduce memory consumption on phones
```

Reply:
This is ugly. I think the property should reflect the configuration, but then we'd need to be able to override it when needed.